src/main/java/io/tiledb/spark/TileDBDataWriter.java

```diff
@@ -219,6 +237,11 @@ private boolean writeRecordToBuffer(
         return true;
       }
     }
+
+    if (record.isNullAt(ordinal)) {
```

Comment:
Could we skip the rest of the processing in case a value is `null`? Perhaps something like:

```
if (record.isNullAt(ordinal)) {
      validityByteMap[bufferElement] = 0;
      nativeArrayBufferElements[bufferIdx] += 1;
      return false;
}
```

Reply:
Done

Reply:
Even though the tests run locally after this change, the CI fails. I am not exactly sure why though.... For now, I reverted back to: 

```
if (record.isNullAt(ordinal)) {
      validityByteMap[bufferElement] = 0;
}
```